.github/workflows/compile-examples.yml

```diff
@@ -0,0 +1,265 @@
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - libraries/SDU/examples
+              - ~/Arduino/libraries/WiFiNINA/examples
+          # mkrnb1500 boards
+          - board:
+              type: "mkrNB1500"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - libraries/SDU/examples
+          # nano_33_iot
+          - board:
+              type: "nano33IoT"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+              - ~/Arduino/libraries/WiFiNINA/examples
+          # mkrwan1310
+          - board:
+              type: "mkrwan1310"
+            additional-sketch-paths: |
+              - libraries/I2S/examples
+          # adafruit_cicrcuitplayground board
+          - board:
+              type: "adafruit_playg"
+            additional-sketch-paths: |
+              - libraries/SDU/examples
+          # mzero boards
+          - board:
+              type: "mzero"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataPlus
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataEthernet
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+              - libraries/SDU/examples
+          # tian board
+          - board:
+              type: "tian"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataPlus
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataEthernet
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+          # MKRGSM1400 board
+          - board:
+              fqbn: "arduino:samd:mkrgsm1400"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/MKRGSM/examples
+          # MKRNB1500 board
+          - board:
+              fqbn: "arduino:samd:mkrnb1500"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/MKRNB/examples
+          # MKRWAN board
+          - board:
+              fqbn: '"arduino:samd:mkrwan1300" "arduino:samd:mkrwan1310"'
```

Comment:
Since none of the existing matrix items match the `matrix.include` filter, this creates a new matrix element.
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-including-new-combinations
You can see the `fqbn: '"arduino:samd:mkrwan1300" "arduino:samd:mkrwan1310"'` job here;
https://github.com/giulcioffi/ArduinoCore-samd/runs/1263161214
and the jobs for the `fqbn: arduino:samd:mkrwan1300` and `fqbn: arduino:samd:mkrwan1310` matrix elements:
https://github.com/giulcioffi/ArduinoCore-samd/runs/1263161020
https://github.com/giulcioffi/ArduinoCore-samd/runs/1263161034

Apparently the `arduino/compile-sketches` action accepts the input `fqbn: "arduino:samd:mkrwan1300" "arduino:samd:mkrwan1310"`and ends up just compiling only for the MKR WAN 1300.

You could add a general "WAN" attribute property to the matrix, something like what I did in this workflow:
https://github.com/arduino/arduino-examples/blob/main/.github/workflows/compile-examples.yml
The only problem is that, as you see here, when no matrix element matches the filter, one is created, which is why I had to define those properties for all the boards in the matrix, rather than only for the boards the property applied to.

You could also just add a separate `matrix.include` filter for each of the MKR WAN boards `board.fqbn` values.

---
This also has the same issue as I described above for `arduino:samd:mkrgsm1400` of overwriting the previous value of `additional-sketch-paths` for `arduino:samd:mkrwan1300`.